redbeams/src/main/java/com/sequenceiq/redbeams/service/stack/RedbeamsUpgradeService.java

```diff
@@ -49,12 +66,27 @@ public void upgradeDatabaseServer(String crn, UpgradeDatabaseRequest upgradeData
             return;
         }
 
+        dbStack = updateNetworkSubnets(dbStack);
         dbStackStatusUpdater.updateStatus(dbStack.getId(), DetailedDBStackStatus.UPGRADE_REQUESTED);
         RedbeamsStartUpgradeRequest redbeamsStartUpgradeRequest = new RedbeamsStartUpgradeRequest(dbStack.getId(),
                 upgradeDatabaseRequest.getTargetMajorVersion());
         flowManager.notify(RedbeamsUpgradeEvent.REDBEAMS_START_UPGRADE_EVENT.selector(), redbeamsStartUpgradeRequest);
     }
 
+    private DBStack updateNetworkSubnets(DBStack dbStack) {
+        String cloudPlatform = dbStack.getCloudPlatform();
+        if (cloudParameterCache.isDbSubnetsUpdateEnabled(cloudPlatform)) {
+            DetailedEnvironmentResponse environment = environmentService.getByCrn(dbStack.getEnvironmentId());
+            Map<String, Object> envSubnets = networkBuilderService.getSubnetsFromEnvironment(environment, CloudPlatform.valueOf(cloudPlatform), dbStack);
+            Map<String, Object> networkAttributes = dbStack.getNetwork().getAttributes().getMap();
+            networkAttributes.putAll(envSubnets);
+            dbStack.getNetwork().setAttributes(new Json(networkAttributes));
+            return dbStackService.save(dbStack);
```

Comment:
Please add a log line that network attributes were updated with env subnets, and the names could also be listed.